packages/tfchain_client/src/utility.ts

```diff
@@ -13,9 +14,16 @@ class Utility {
   async batch<T>(extrinsics: ExtrinsicResult<T>[]): Promise<T[]> {
     extrinsics = extrinsics.filter(Boolean);
     if (extrinsics.length > 0) {
-      const { resultSections, resultEvents } = this.extractResultSectionsAndEvents(extrinsics);
-      const batchExtrinsic = await this.client.api.tx.utility.batch(extrinsics);
-      return this.client.applyExtrinsic<T[]>(batchExtrinsic, resultSections, resultEvents);
+      const result: T[] = [];
+      for (let i = 0; i < extrinsics.length; i += batchSize) {
+        const batch = extrinsics.slice(i, i + batchSize);
+        const { resultSections, resultEvents } = this.extractResultSectionsAndEvents(batch);
+        const batchExtrinsic = await this.client.api.tx.utility.batch(batch);
+        const res = await this.client.applyExtrinsic<T[]>(batchExtrinsic, resultSections, resultEvents);
+        result.concat(res);

```

Comment:
we should overwrite the variable again as `concat` returns the concatenation result and doesn't change the list.
![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/41957921/4f9817ff-5b22-4099-a6bd-c4eeb2ca5776)

```suggestion
        result = result.concat(res);

```